Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Build stage
+FROM python:3.10-slim AS build
+
+# Set work directory
+WORKDIR /app
+
+# Copy only requirements.txt to install dependencies
+COPY requirements.txt .
+
+# Install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy application files
+COPY . .
+
+# Run stage
+FROM python:3.10-slim AS run
+
+# Set environment variables
+ENV PYTHONUNBUFFERED=1
+
+# Set work directory
+WORKDIR /app
+
+# Copy installed dependencies from the build stage
+COPY --from=build /usr/local/lib/python3.10/site-packages /usr/local/lib/python3.10/site-packages
```

Comment:
Dependency isolation is not correctly implemented. Ensure that only the necessary dependencies are copied from the build stage to the run stage to prevent unexpected behavior.